### PR TITLE
Use pg14-based timescaledb images in docker-compose

### PR DESCRIPTION
**Pull request by JamesGuthrie**

### CODE REVIEW

**Review by paulfantom — APPROVED** [review]
lgtm

File: docker-compose/high-availability/docker-compose.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: '3.0'
 
 services:
   db:
-    image: timescaledev/promscale-extension:latest-pg12
+    image: timescaledev/promscale-extension:latest-ts2-pg14
```
Comment (cevian) [inline docker-compose/high-availability/docker-compose.yaml:5]: lets actually change this to `timescale/timescaledb-ha:pg14-latest`.

**Review by cevian — APPROVED** [review]
Lets put this as a not in the changelog as a breaking change. People can't just switch to this docker compose from the old docker compose without manual intervention.

File: docker-compose/high-availability/docker-compose.yaml (hunk 1 above)
Comment (bitwalker) [reply docker-compose/high-availability/docker-compose.yaml:5]: I think we need to produce an HA version of the extension image and use that.

File: docker-compose/docker-compose.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: '3.0'
 
 services:
   db:
-    image: timescaledev/timescaledb-ha:pg12-latest
+    image: timescale/timescaledb-ha:pg14-latest
```
Comment (bitwalker) [inline docker-compose/docker-compose.yaml:5]: Shouldn't this also use the promscale-extension image? Especially now that we're planning to require the extension, might be good to already have these files depend on that image.

Comment (JamesGuthrie) [reply docker-compose/docker-compose.yaml:5]: timescaledb-ha also has the extension packaged.

**Comment by JamesGuthrie** [comment]
@cevian 

> Lets put this as a not in the changelog as a breaking change. People can't just switch to this docker compose from the old docker compose without manual intervention.

Technically they can. The docker-compose doesn't define volumes or volume mounts which would be reused over invocations of docker-compose, so changing the docker image shouldn't actually affect anything or anyone.

File: docker-compose/high-availability/docker-compose.yaml (hunk 1 above)
Comment (JamesGuthrie) [reply docker-compose/high-availability/docker-compose.yaml:5]: @cevian Do you know why we were not using timescaledb-ha previously?

Comment (cevian) [reply docker-compose/high-availability/docker-compose.yaml:5]: @bitwalker the extension should already be in the HA image

Comment (cevian) [reply docker-compose/high-availability/docker-compose.yaml:5]: @JamesGuthrie I am guessing legacy oversight

File: docker-compose/docker-compose.yaml (hunk 2 above)
Comment (cevian) [reply docker-compose/docker-compose.yaml:5]: Yep its already there

**Review by cevian — COMMENTED** [review]
I'll approve this change once we switch over to the ha image

File: docker-compose/high-availability/docker-compose.yaml (hunk 1 above)
Comment (JamesGuthrie) [reply docker-compose/high-availability/docker-compose.yaml:5]: done

**Pull request merged**